File: 프로그래머스/코딩테스트_고득점_kit/해시/위장/dailyco.js
/*
작성자 : dailyco
문제 난이도: 하
해결시간: 00:20:00
문제 링크 : https://programmers.co.kr/learn/courses/30/lessons/42578
블로그 링크 : https://dailyco.github.io/posts/camouflage/
*/

function solution(clothes) {
  var obj = {};
  var answer = 1;

  for (let c of clothes) obj[c[1]] = (obj[c[1]] || 0) + 1;

  for (let key in obj) answer *= obj[key] + 1;

  return answer - 1;
}

File: 프로그래머스/코딩테스트_고득점_kit/동적계획법/정수_삼각형/dailyco.cpp
/*
작성자 : dailyco
문제 난이도: 중
해결시간: 00:30:00
문제 링크 : https://programmers.co.kr/learn/courses/30/lessons/43105
블로그 링크 : https://dailyco.github.io/posts/integer_triangle/
*/

#include <vector>

using namespace std;

int solution(vector<vector<int>> triangle)
{
    for (int i = triangle.size() - 2; i >= 0; i--)
    {
        for (int j = 0; j <= i; j++)
        {
            if (triangle[i][j] + triangle[i + 1][j] >= triangle[i][j] + triangle[i + 1][j + 1])
                triangle[i][j] += triangle[i + 1][j];
            else
                triangle[i][j] += triangle[i + 1][j + 1];
        }
    }

    return triangle[0][0];
}